puppet/lib/puppet/reports/puppetdb.rb

```diff
@@ -48,31 +48,41 @@ def report_to_hash
         "producer_timestamp"      => Puppet::Util::Puppetdb.to_wire_time(Time.now),
         "start_time"              => Puppet::Util::Puppetdb.to_wire_time(time),
         "end_time"                => Puppet::Util::Puppetdb.to_wire_time(time + run_duration),
-        "resource_events"         => resource_events,
         "environment"             => environment,
         "transaction_uuid"        => transaction_uuid,
         "status"                  => status,
         "noop"                    => is_noop,
         "logs"                    => build_logs_list,
         "metrics"                 => build_metrics_list,
+        "resources"               => resources,
       }
     end
   end
 
+  # @return TrueClass
+  # @api private
+  def has_noop_event?(resource)
+    resource["events"].any? { |event| event["status"] == 'noop' }
+  end
+
+  # @return TrueClass
+  # @api private
+  def has_failed_event?(resource)
+    resource["events"].any? { |event| event["status"] == 'failed' }
+  end
+
   # @return Array[Hash]
   # @api private
-  def build_events_list
-    profile("Build events list (count: #{resource_statuses.count})",
-            [:puppetdb, :events_list, :build]) do
-      filter_events(resource_statuses.inject([]) do |events, status_entry|
-        _, status = *status_entry
-        if ! (status.events.empty?)
-          events.concat(status.events.map { |event| event_to_hash(status, event) })
-        elsif status.skipped
-          events.concat([fabricate_event(status, "skipped")])
-        end
-        events
-      end)
+  def build_resources_list
+    profile("Build resources list (count: #{resource_statuses.count})",
+            [:puppetdb, :resources_list, :build]) do
+      include_unchanged_resources = false
+      resources = resource_statuses.values.map { |resource| resource_status_to_hash(resource) }
+      if ! include_unchanged_resources
```

Comment:
Nope it's another ticket to make it a configurable value, I just named it for now to make that ticket easier when the time comes!